### PR TITLE
feat(#260): Add Integration Test For High Level Transformation

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
I added integration test that shows what we want to get as a result of decompilation bytecode into EO. Here is only the high-level overview and a lot of details are omitted.

Closes: #260.
____
History:
- feat(#260): Add integration test skeleton and expected transformations
- feat(#260): fix all qulice suggestions

---

## PR-Codex overview
This PR focuses on adding a new dependency and test resources to the project. 

### Detailed summary
- Added a new dependency `org.eolang:jucs:0.2.0` in the `pom.xml` file.
- Added a new test resource file `simple_objects.yaml` in the `src/test/resources/packs` directory.
- Added a new Java class `Application.java` in the `src/test/resources/packs` directory.
- Added a new Java class `A.java` in the `src/test/resources/packs` directory.
- Added a new Java class `B.java` in the `src/test/resources/packs` directory.
- Added a new EO file `Application.eo` in the `src/test/resources/packs` directory.
- Added a new EO file `A.eo` in the `src/test/resources/packs` directory.
- Added a new EO file `B.eo` in the `src/test/resources/packs` directory.
- Added a new Java test class `TrasformationPacksTest.java` in the `src/test/java/it` directory.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Is it what we want to receive as a result of bytecode decompilation?

File: src/test/resources/packs/simple_objects.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    package com.example
+    [] > Application
+      [args...] > main
+        seq > @
+          (b (a 42).new).new.bar
+  A.eo: |
+    package com.example
+    [] > A
+      cage x   
+      [xa] > new
```
Comment (maxonfjvipon) [inline src/test/resources/packs/simple_objects.yaml:42]: @volodya-lombrozo `xa` should be a free argument of `A`. You may see in `main` that you do: `(a 42).new` (42 goes to `a`, not to `a.new`). If you leave them as args of `new`, then code in `main` should look like
`(b.new (a.new 42)).bar`. 
`(b (a 42).new).new.bar` seems more correct if we talk about java -> eo mapping

File: src/test/resources/packs/simple_objects.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        x * 2 > @
+  B.eo: |
+    package com.example
+    [] > B
+      cage a
+      [aa] > new
+        seq > @
+          a.write aa
+      [] > bar
+        (a.foo + 1) > @
```
Comment (maxonfjvipon) [inline src/test/resources/packs/simple_objects.yaml:55]: @volodya-lombrozo I think `+` and `*` should be mapped to `eo` objects like `plus` and `times`

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: src/test/resources/packs/simple_objects.yaml (hunk 2 above)
Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml:55]: @maxonfjvipon agree

File: src/test/resources/packs/simple_objects.yaml (hunk 1 above)
Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml:42]: @maxonfjvipon What if `A` will have 2 different constructors with different number of parameters? In this case we will have a problem

Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml:42]: I mean if we leave `xa` as an argument for for `A`

Comment (maxonfjvipon) [reply src/test/resources/packs/simple_objects.yaml:42]: @volodya-lombrozo you're right. Let's stick with arguments in `new`. Then example in `main` should look like: `(b.new (a.new 42)).bar`

Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml:42]: @maxonfjvipon Done

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you take a look one more time, please?

File: src/test/resources/packs/simple_objects.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  Application.eo: |
+    package com.example
+    [] > Application
+      [args...] > main
+        seq > @
+          (b.new (a.new 42)).bar
+  A.eo: |
+    package com.example
+    [] > A
+      cage x
```
Comment (maxonfjvipon) [inline src/test/resources/packs/simple_objects.yaml:41]: @volodya-lombrozo here it should be `cage > x`

File: src/test/resources/packs/simple_objects.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      cage x   
+      [xa] > new
+        seq > @
+          x.write xa 
+      [] > foo
+        times x 2 > @
+  B.eo: |
+    package com.example
+    [] > B
+      cage a
```
Comment (maxonfjvipon) [inline src/test/resources/packs/simple_objects.yaml:50]: @volodya-lombrozo here it should be `cage > a`

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: src/test/resources/packs/simple_objects.yaml (hunk 3 above)
Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml]: @maxonfjvipon Done

File: src/test/resources/packs/simple_objects.yaml (hunk 4 above)
Comment (volodya-lombrozo) [reply src/test/resources/packs/simple_objects.yaml]: @maxonfjvipon Done

**Review by maxonfjvipon — APPROVED** [review]
@volodya-lombrozo I'm not sure about `plus a.foo 1` instead of `a.foo.plus 1`, but I believe it would not a big problem

**Review by maxonfjvipon — APPROVED** [review]
@volodya-lombrozo I think it's good enough

**Comment by volodya-lombrozo** [comment]
@rultor merge

**Pull request merged**